packages/expo-av/android/src/main/java/expo/modules/av/progress/ProgressLooper.kt

```diff
@@ -0,0 +1,74 @@
+package expo.modules.av.progress
+
+typealias TimeMachineTick = () -> Unit
+
+interface TimeMachine {
+  fun scheduleAt(intervalMillis: Long, callback: TimeMachineTick)
```

Comment:
I was wondering why have you chosen such name and parameters order, but it's to allow such syntax:?
```kt
timeMachine.scheduleAt(nextExpectedTick - timeMachine.time) {
  waiting = false
  listener?.invoke()
  scheduleNextTick()
}
```

Reply:
I generally like to put callbacks at the end. Such syntax is one of main reasons. Another is, that I find this code hard to read:
```
new Handler.postDelayed(() -> {
  // some code here
  // possibly multiple lines
}, 500); // And here the actual delay.
```